polaris-react/src/components/Avatar/Avatar.tsx

```diff
@@ -19,6 +19,27 @@ enum Status {
 
 export const STYLE_CLASSES = ['one', 'two', 'three', 'four', 'five'];
 
+/**
+ * Computes a rudimentary hash from a string by xoring the character codes
+ * of all characters
+ * @param str A string
+ */
+function xorHash(str: string) {
+  let hash = 0;
+
+  for (const char of str) {
+    hash ^= char.charCodeAt(0);
+  }
+
+  return hash;
+}
+
+function styleClass(name?: string) {
```

Comment:
There was no reason to define this function inside the component, so moved it out